### PR TITLE
Adding activities page

### PR DESCRIPTION
**Pull request by shailee-m**
- Adding  activities page
- Updating manage activity component to support opening in edit mode
- Moving adding the cookie to API in Axios interceptor. Also, moving the code to disable cache to Axios interceptor

### CODE REVIEW

**Comment by shailee-m** [comment]
Requires #2 to be completed to complete this

File: frontend/src/components/ManageActivity.vue [hunk 1]
```diff
@@ -110,6 +115,21 @@ export default {
       duration: 0
     };
   },
+
+  async created() {
+    await this.$nextTick();
```
Comment (brylie) [inline frontend/src/components/ManageActivity.vue]: Why do we `await this.$nextTick()` twice here?

File: frontend/src/layouts/MainLayout.vue [hunk 2]
```diff
@@ -18,8 +18,15 @@ import GeriLifeTopbar from "components/GeriLifeTopbar.vue";
 export default {
   name: "MainLayout",
   components: { GeriLifeTopbar },
-  data() {
-    return {};
+  async created() {
+    const { data: cookieData } = await this.$axios.post("/users/login", {
+      email: "mehtashailee21@gmail.com",
+      password: "123456"
+    });
```
Comment (brylie) [inline frontend/src/layouts/MainLayout.vue:25]: Should we create a login form soon?

**Comment by brylie** [comment]
This looks good overall. Perhaps it would be helpful to have some instructions on how this project works with the backend, i.e. how to get set up. Also, let's get a basic login form ready soon, even if it is hard-coded to use localhost/port for Meteor instance.

File: frontend/src/components/ManageActivity.vue (hunk 1 above)
Comment (shailee-m) [reply frontend/src/components/ManageActivity.vue]: Seems to be an issue with Quasar-vue framework, props are not immediately available in the created hook. Need to wait for 2 cycles for data to be updated. Will look for any other solutions if possible

File: frontend/src/layouts/MainLayout.vue (hunk 2 above)
Comment (shailee-m) [reply frontend/src/layouts/MainLayout.vue:25]: Yes definitely. I will work on it next

**Comment by shailee-m** [comment]
> This looks good overall. Perhaps it would be helpful to have some instructions on how this project works with the backend, i.e. how to get set up. Also, let's get a basic login form ready soon, even if it is hard-coded to use localhost/port for Meteor instance.

Will add the instructions in this PR.

**Comment by shailee-m** [comment]
> This looks good overall. Perhaps it would be helpful to have some instructions on how this project works with the backend, i.e. how to get set up. Also, let's get a basic login form ready soon, even if it is hard-coded to use localhost/port for Meteor instance.

Done

File: frontend/src/components/ManageActivity.vue (hunk 1 above)
Comment (brylie) [reply frontend/src/components/ManageActivity.vue]: OK, no worries. Please just add a code comment with a link to the up-stream issue report.

**Comment by brylie** [comment]
@shailee-m is this ready for review again or are you still working on anything in this branch?

File: frontend/src/components/ManageActivity.vue (hunk 1 above)
Comment (shailee-m) [reply frontend/src/components/ManageActivity.vue]: No need. I added a v-if in the calling component

**Comment by shailee-m** [comment]
> @shailee-m is this ready for review again or are you still working on anything in this branch?

It is ready for review

**Comment by brylie** [comment]
@shailee-m I am getting the following error when loading the front page:

> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:3000/users/login?_t=1600767557679. (Reason: header ‘access-control-allow-headers’ is not allowed according to header ‘Access-Control-Allow-Headers’ from CORS preflight response).

I can fix the issue temporarily by editing startup.js in the wellbeing app and changing the `Access-Control-Allow-Headers` value to `*`

```js
'Access-Control-Allow-Headers': '*',
```

**Comment by brylie** [comment]
Disregard my previous message. My branch was out of sync.

**Pull request merged**